test/widgets/dialog_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/dialog.dart';
+
+import '../model/binding.dart';
+import 'dialog_checks.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  late BuildContext context;
+
+  const String title = "Dialog Title";
+  const String message = "Dialog message.";
+
+  Future<void> prepare(WidgetTester tester) async {
+    addTearDown(testBinding.reset);
+
+    await tester.pumpWidget(const TestZulipApp(
+      child: Scaffold(body: Placeholder())));
+    await tester.pump();
+    context = tester.element(find.byType(Placeholder));
+  }
+
+  group('showErrorDialog', () {
+    testWidgets('show error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+      checkErrorDialog(tester, expectedTitle: title, expectedMessage: message);
+
+    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
+
+    testWidgets('user closes error dialog', (tester) async {
+      await prepare(tester);
+
+      showErrorDialog(context: context, title: title, message: message);
+      await tester.pump();
+
+      Widget button = checkErrorDialog(tester, expectedTitle: title);
+      await tester.tap(find.byWidget(button));
+      await tester.pump();
+      checkNoDialog(tester);
```

Comment:
In the current version, the analyzer fails here after the first or second commit. A PR should pass all tests at each commit of the PR, in order to keep the commits fully coherent:
https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#each-commit-must-be-coherent

Probably cleanest is to have the rename of `checkNoDialog`, and accompanying changes, happen as the first commit.